specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,567 @@
+    notification.Tag(L"Tag");
+    notification.Group(L"Group");
+    notification.ExpiresOnReboot(true);
+    AppNotificationManager::Default().Show(notification);
+}
+```
+## Processing an Activation
+
+The app will need to process the activator in response to a User interacting with the notification in Action Centre. 2 common
+scenarios here are
+
+1. Have the app launch in the foreground in a specific UI context OR
+2. Have the app process an action specific behavior (like a button press in the payload body) in the background.
+
+```cpp
+//<toast launch="action=openThread&amp;threadId=92187">
+//  <visual>
+//      <binding template="ToastGeneric">
+//          <text hint-maxLines="1">Jill Bender</text>
+//          <text>Check out where we camped last weekend! Incredible!!</text>
+//          <image placement="hero" src="https://unsplash.it/360/180?image=1043" />
+//      </binding>
+//  </visual>
+//  <actions>
+//      <input id="textBox" type="text" placeHolderContent="reply" / >
+//      <action
+//          content="Send"
+//          imageUri="Assets/Icons/send.png"
+//          hint-inputId="textBox"
+//          activationType="background"
+//          arguments="action=reply&amp;threadId=92187" />
+//  </actions>
+//</toast>
+
+const winrt::hstring c_notificationLaunchAction = L"action=openThread&amp;threadId=92187";
+const winrt::hstring c_notificationReplyButtonAction = L"action=reply&amp;threadId=92187";
+
+void ProcessNotificationArgs(const AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+{
+    if (notificationActivatedEventArgs.ActivationArgs() == c_notificationLaunchAction)
+    {
+        // The user clicks on the notification: So use the launchAction to do stuff
+        // Do LaunchAction Stuff
+    }
+    else if (notificationActivatedEventArgs.ActivationArgs() == c_notificationReplyButtonAction)
+    {
+        // The user clicked on the reply button on the notification. So query the input field
+        auto input = notificationActivatedEventArgs.UserInput();
+        auto replyBoxText = input.Lookup(L"ReplyBox");
+
+        // Process the reply text
+        ProcessReply(replyBoxText);
+    }
+}
+```
+
+## Processing Active Notifications
+
+We define active notifications as payloads that have already been displayed by the app and show up in the
+Action Centre. We can perform a number of operations on these notifications such as retrieval and
+deletion.
+
+```cpp
+// Get a List of all active notifications displayed in the Action Centre
+winrt::Windows::Foundation::IAsyncOperation<Windows::Foundation::Collections::IVector<AppNotification>> GetNotificationListASync()
+{
+    auto notifications = co_await AppNotificationManager::Default().GetAllAsync();
+    co_return notifications;
+}
+
+// Remove a notification from action centre using tag and group
+winrt::Windows::Foundation::IAsyncAction RemoveNotification(const winrt::hstring& tag, const winrt::hstring& group)
```

Comment:
dangerous use of & param in async method

Reply:
Removed &